docs/se/guides/40_webclient.adoc

```diff
@@ -0,0 +1,420 @@
+
+In practice, String is not the most useful return type, since it usually needs some more handling.In this case it could
+be more interesting to return an object such as JsonObject. Earlier, Json support was added to the webclient,
+so it can be used instead of String.
+
+[source,java]
+.Replace String by JsonObject:
+----
+import javax.json.JsonObject;
+
+...
+
+webClient.get()
+    .path("/greet/David")
+    .request(JsonObject.class)  // <1>
+    .peek(System.out::println)
+    .await();
+----
+<1> Request a JsonObject as return value.
+
+In the Uri, the String following `greet` is a path parameter which allows the application to greet someone.
+
+[source,bash]
+.Output:
+----
+{"message":"Hello David!"}
+----
+
+This time, a JsonObject is printed out in the console. It is possible to change the greeting itself by using a PUT
+request to `/greet/greeting` endpoint from the base Uri.
+
+[source,java]
+.Modify the application greeting:
+----
+import javax.json.Json;
+
+...
+
+JsonObject entity = Json.createObjectBuilder() // <1>
+    .add("greeting", "Bonjour")
+    .build();
+webClient.put()     // <2>
+    .path("/greet/greeting")
+    .submit(entity)     // <3>
+    .thenCompose(response -> webClient.get()    // <4>
+        .path("/greet/David")
+        .request(JsonObject.class))
+    .thenAccept(System.out::println)
+    .await();
+----
+<1> Create a JsonObject with key `greeting` and value `bonjour`.
+<2> Create a PUT request.
+<3> Submit the JsonObject created earlier.
+<4> Once done, make a GET call to verify the modification was processed to the greeting.
+
+According to the quickstart documentation, a Json object can be sent to the application to change the greeting following
+this structure: `{"greeting" : "value"}`. The first three lines of code create the JsonObject with the required content.
+This time, we use the PUT request and submit methods to push the new greeting. One way to check the greeting modification
+is to execute GET request again and display obtained response. `thenCompose` method will execute a GET request after
+the PUT request is executed.
+
+[source,bash]
+.Output:
+----
+{"message":"Bonjour David!"}
+----
+
+=== WebClient Metrics
+
+WebClient as other Helidon components support Metrics. This chapter introduces the different metrics usable to measure
+the web client activity. There are two ways to set up metrics, first one is programmatically on the WebClient instance,
+second one is to use configuration file.
+
+==== Add metrics dependency
+
+There is a specific dependency to use web client metrics in your application.
+
+[source,xml]
+.Add the following dependency to pom.xml:
+----
+<dependency>
+    <groupId>io.helidon.webclient</groupId>
+    <artifactId>helidon-webclient-metrics</artifactId>
+</dependency>
+----
+
+==== Set up metrics on WebClient instance
+
+It is possible to register metrics on WebClient directly into the code. Will be described a general method to use with
```

Comment:
It is possible to register metrics on WebClient directly into the code. The following example shows a general method that can be used with any metric.